gcp_variant_transforms/libs/bigquery_vcf_schema_converter.py

```diff
@@ -222,24 +215,90 @@ def _add_format_fields(schema, formats):
   for field in schema.fields:
     if field.name in _CONSTANT_CALL_FIELDS:
       continue
-    formats.update({
-        field.name: _Format(field.name,
-                            parser.field_counts[vcfio.MISSING_FIELD_VALUE],
-                            _BIG_QUERY_TYPE_TO_VCF_TYPE_MAP.get(field.type),
-                            field.description)})
+    elif field.name in vcf_reserved_fields.FORMAT_FIELDS.keys():
+      reserved_definition = vcf_reserved_fields.FORMAT_FIELDS.get(field.name)
+      _validate_reserved_field(field, reserved_definition)
+      formats.update({field.name: reserved_definition})
+    else:
+      formats.update({field.name: _Format(
+          field.name,
+          bigquery_util.get_vcf_num_from_bigquery_schema(field.mode,
+                                                         field.type),
+          bigquery_util.get_vcf_type_from_bigquery_type(field.type),
+          field.description)})
+
+
+def _add_info_fields(field, infos):
+  # type: (bigquery.TableFieldSchema, Dict[str, _Info]) -> None
+  if field.name == bigquery_util.ColumnKeyConstants.ALTERNATE_BASES:
+    _add_info_fields_from_alternate_bases(field, infos)
+  elif field.name in vcf_reserved_fields.INFO_FIELDS.keys():
+    reserved_definition = vcf_reserved_fields.INFO_FIELDS.get(field.name)
+    _validate_reserved_field(field, reserved_definition)
+    infos.update({field.name: reserved_definition})
+  else:
+    infos.update({field.name: _Info(
+        field.name,
+        bigquery_util.get_vcf_num_from_bigquery_schema(field.mode, field.type),
+        bigquery_util.get_vcf_type_from_bigquery_type(field.type),
+        field.description,
+        None,
+        None)})
 
 
 def _add_info_fields_from_alternate_bases(schema, infos):
   # type: (bigquery.TableFieldSchema, Dict[str, _Info]) -> None
+  """Adds schema nested fields in alternate bases to `infos`.
+
+  Notice that the validation of field mode is skipped for reserved fields since
+  the mode (NULLABLE) of field in alternate bases is expected to be different
+  from the mode (REPEATED) in reserved field definition.
+  """
   for field in schema.fields:
     if field.name in _CONSTANT_ALTERNATE_BASES_FIELDS:
       continue
+    elif field.name in vcf_reserved_fields.INFO_FIELDS.keys():
+      reserved_definition = vcf_reserved_fields.INFO_FIELDS.get(field.name)
+      _validate_reserved_field_type(field, reserved_definition)
```

Comment:
I think this may fail as the mode for all these will be NULLABLE, but we expect REPEATED (i.e. from `Number=A`)? Perhaps you can just call `_validate_reserved_field_type` here.

Reply:
Yeah, I only checked the type here.